src/Illuminate/Collections/Collection.php

```diff
@@ -921,6 +951,32 @@ public function only($keys)
         return new static(Arr::only($this->items, $keys));
     }
 
+    /**
+     * Run a map and reject over each of the items.
+     *
+     * @template TMapValue
+     *
+     * @param  callable(TValue, TKey): TMapValue  $callback
+     * @param  (callable(TValue, TKey): bool)  $reject
+     * @return static<TKey, TMapValue>
+     */
+    public function rejectMap($callable, $reject)
+    {
+        if ((new ReflectionFunction($reject))->getNumberOfParameters() === 1) {
+            $reject = fn ($value) => $reject($value);
+        }
+
+        $result = [];
+
+        foreach ($this->items as $key => $value) {
```

Comment:
Can't we just reuse filterMap here? Like `$this->filterMap($callable, fn ($v) => ! $reject($v))`

Reply:
No we cannot as we need to be able to check the parameter count of the original function.

The tests fail without this work.

Reply:
Well it seems that `return $this->filterMap($callable, (new ReflectionFunction($reject))->getNumberOfParameters() === 1 ? fn ($v) => ! $reject($v) : fn ($v, $k) => ! $reject($v, $k))`

Would do the trick no ?